swift/apple/FirezoneKit/Sources/FirezoneKit/Views/MenuBar.swift

```diff
@@ -236,176 +164,164 @@ public final class MenuBar: NSObject, ObservableObject {
     return menuItem
   }()
 
-  private func createMenu() {
-    menu.addItem(signInMenuItem)
-    menu.addItem(signOutMenuItem)
-    menu.addItem(NSMenuItem.separator())
-
-    menu.addItem(resourcesTitleMenuItem)
-    menu.addItem(resourcesUnavailableMenuItem)
-    menu.addItem(resourcesUnavailableReasonMenuItem)
-    menu.addItem(resourcesSeparatorMenuItem)
-
-    if !store.favorites.isEmpty() {
-      menu.addItem(otherResourcesMenuItem)
-      menu.addItem(otherResourcesSeparatorMenuItem)
-    }
+  public init(store: Store) {
+    statusItem = NSStatusBar.system.statusItem(withLength: NSStatusItem.variableLength)
+    self.store = store
+    self.signedOutIcon = NSImage(named: "MenuBarIconSignedOut")
+    self.signedInConnectedIcon = NSImage(named: "MenuBarIconSignedInConnected")
+    self.signedOutIconNotification = NSImage(named: "MenuBarIconSignedOutNotification")
+    self.signedInConnectedIconNotification = NSImage(named: "MenuBarIconSignedInConnectedNotification")
+    self.siteOnlineIcon = NSImage(named: NSImage.statusAvailableName)
+    self.siteOfflineIcon = NSImage(named: NSImage.statusUnavailableName)
+    self.siteUnknownIcon = NSImage(named: NSImage.statusNoneName)
+    self.connectingAnimationImages[.first] = NSImage(named: "MenuBarIconConnecting1")
+    self.connectingAnimationImages[.second] = NSImage(named: "MenuBarIconConnecting2")
+    self.connectingAnimationImages[.last] = NSImage(named: "MenuBarIconConnecting3")
 
-    menu.addItem(aboutMenuItem)
-    menu.addItem(adminPortalMenuItem)
-    menu.addItem(helpMenuItem)
-    menu.addItem(settingsMenuItem)
-    menu.addItem(NSMenuItem.separator())
-    menu.addItem(quitMenuItem)
+    super.init()
 
-    menu.delegate = self
-    statusItem.menu = menu
+    updateStatusItemIcon()
+    createMenu()
+    setupObservers()
   }
 
-  private func createMenuItem(
-    _: NSMenu,
-    title: String,
-    action: Selector?,
-    isHidden: Bool = false,
-    key: String = "",
-    target: AnyObject?
-  ) -> NSMenuItem {
-    let item = NSMenuItem(title: title, action: action, keyEquivalent: key)
-
-    item.isHidden = isHidden
-    item.target = target
-    item.isEnabled = (action != nil)
+  // MARK: Responding to state updates
```

Comment:
What is a `// MARK` comment?

Reply:
It's a comment that marks a certain section of the file so you can jump to it more easily in Xcode. It's pretty common in many Swift codebases.